src/char/int_storage.c

```diff
@@ -41,24 +41,144 @@ struct inter_storage_interface inter_storage_s;
+					StrBuf->Printf(&buf, ", `expire_time`, `bound`, `unique_id`) VALUES");
+				}
+
+				StrBuf->Printf(&buf, "('%d', '%d', '%d', '%u', '%d', '%d', '%d'",
+							   cp_it->id, account_id, p_it->amount, p_it->equip, p_it->identify, p_it->refine, p_it->attribute);
+				for (k = 0; k < MAX_SLOTS; k++)
+					StrBuf->Printf(&buf, "'%d'", p_it->card[k]);
+				for (k = 0; k < MAX_ITEM_OPTIONS; ++k)
+					StrBuf->Printf(&buf, "'%d', '%d'", p_it->option[k].index, p_it->option[k].value);
+				StrBuf->Printf(&buf, ", '%u', '%d', '%"PRIu64"')%s", p_it->expire_time, p_it->bound, p_it->unique_id, total_updates > 0 ? ", " : "");
+
+				total_updates++;
+				updated_p[j] = true;
+			}
+		} else { // Does not exist, so set for deletion.
+			delete[total_deletes++] = cp_it->id;
+		}
+	}
+
+	if (total_updates > 0 && SQL_ERROR == SQL->QueryStr(inter->sql_handle, StrBuf->Value(&buf)))
+		Sql_ShowDebug(inter->sql_handle);
+
+	/**
+	 * Handle deletions, if any.
+	 */
+	if (total_deletes > 0) {
+		StrBuf->Clear(&buf);
+		StrBuf->Printf(&buf, "DELETE FROM `%s` WHERE `id` IN (", storage_db);
+		for (i = 0; i < total_deletes; i++) {
+			StrBuf->Printf(&buf, ", '%d'", delete[i]);
+		}
+		StrBuf->Printf(&buf, ");");
+
+		if (SQL_ERROR == SQL->QueryStr(inter->sql_handle, StrBuf->Value(&buf)))
+			Sql_ShowDebug(inter->sql_handle);
+	}
+
+	/**
+	 * Check for new items.
+	 */
+	for (i = 0; i < VECTOR_LENGTH(p->item); i++) {
+		struct item *p_it = &VECTOR_INDEX(p->item, i);
+
+		if (updated_p[i] == true)
+			continue; // Item already dealt with, lets continue.
+
+		// Store the remaining items.
+		if (total_inserts == 0) {
+			StrBuf->Clear(&buf);
+			StrBuf->Printf(&buf, "INSERT INTO `%s`(`account_id`, `nameid`, `amount`, `equip`, `identify`, `refine`, `attribute`, `expire_time`, `bound`, `unique_id`", storage_db);
+			for (j = 0; j < MAX_SLOTS; ++j)
+				StrBuf->Printf(&buf, ", `card%d`", j);
+			for (j = 0; j < MAX_ITEM_OPTIONS; ++j)
+				StrBuf->Printf(&buf, ", `opt_idx%d`, `opt_val%d`", j, j);
+			StrBuf->AppendStr(&buf, ") VALUES ");
+		}
+
+		StrBuf->Printf(&buf, "('%d', '%d', '%d', '%u', '%d', '%d', '%d', '%u', '%d', '%"PRIu64"'",
+				account_id, p_it->nameid, p_it->amount, p_it->equip, p_it->identify, p_it->refine,
+				p_it->attribute, p_it->expire_time, p_it->bound, p_it->unique_id);
+		for (j = 0; j < MAX_SLOTS; ++j)
+			StrBuf->Printf(&buf, ", '%d'", p_it->card[j]);
+		for (j = 0; j < MAX_ITEM_OPTIONS; ++j)
+			StrBuf->Printf(&buf, ", '%d', '%d'", p_it->option[j].index, p_it->option[j].value);
+		StrBuf->Printf(&buf, ")%s", total_inserts > 0 ? ", " : "");
+
+		total_inserts++;
+	}
+
+	if (total_inserts > 0 && SQL_ERROR == SQL->QueryStr(inter->sql_handle, StrBuf->Value(&buf)))
+		Sql_ShowDebug(inter->sql_handle);
+
+	StrBuf->Destroy(&buf);
+
+	/**
+	 * Sync map-storage with inter-storage
+	 */
+	VECTOR_CLEAR(cp->item);
+	VECTOR_ENSURE(cp->item, VECTOR_LENGTH(p->item), 1);
+	VECTOR_PUSHARRAY(cp->item, VECTOR_DATA(p->item), VECTOR_LENGTH(p->item));
```

Comment:
I think querying the table for a final sync would be better because there is no proper way to get the `id` field of the table after insertion (without splitting the insert queries per item), so I'll go with that.